### ISSUE TITLE
Generate a "client config" with booleans instead of strings.

### ISSUE THREAD

**Issue by mkraft** [mattermost/mattermost-server#16839]
#### Summary

The intent is that the API for the client config return values that are logically boolean as JSON values `true` or `false` rather than string values `"true"` or `"false"`.

#### Steps to reproduce
Request the client config from the API:

```bash
curl 'https://community-daily.mattermost.com/api/v4/config/client?format=old'
```

#### Expected behavior

Values are returned using the same value type as defined in the config models. For example, [`AnnouncementSettings.AllowBannerDismissal` is a `*bool`](https://github.com/mattermost/mattermost-server/blob/9c9fbe6dd9b6a89095eb6692287e5aaa0c0436ae/model/config.go#L1753), so it should match that type when marshalled. 

The new values are returned via:

```
GET 'https://my-domain.test/api/v4/config/client?format=new'
```

#### Possible fixes

This will involve a change to the [`config.GenerateLimitedClientConfig` function](https://github.com/mattermost/mattermost-server/blob/9c9fbe6dd9b6a89095eb6692287e5aaa0c0436ae/config/client.go#L15) and likely could be solved by defining the new `ClientConfig` struct for ease of marshalling, as done in [the `(*Server).regenerateClientConfig` method](https://github.com/mattermost/mattermost-server/blob/9c9fbe6dd9b6a89095eb6692287e5aaa0c0436ae/app/config.go#L323).

This is the server-side portion of https://github.com/mattermost/mattermost-server/issues/16802. 

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

**Comment by compiledsound** [comment]
I can pick this up

**Comment by mkraft** [comment]
> I can pick this up

Awesome @compiledsound, thank you! Please reach out to me here or on https://community-daily.mattermost.com with any questions of if you want to brainstorm or hit any snags or whatever :)

**Comment by compiledsound** [comment]
@hmhealey my approach is to create a struct with all the fields mentioned in the related methods, but it seems that limited client config and the regular client config may need to be separate structs? Unless we expect to change the test cases, i.e. removing tests that check for the existence of fields in the config. Thoughts?

**Comment by hmhealey** [comment]
I could see either defining a proper struct for the config or just defining it as `map[string]interface{}` similar to the current version of the code. On one hand, a proper struct type would be nicer for working with in Go code (like for anyone using `model/client4.go`), but on the other hand, the limited config having a different return type makes that awkward. We could solve that by using pointers to the fields, but then we have to nil check them all which makes them kind of awkward.

I haven't worked on the server in a while, so @mkraft might have a better idea of what would be more idiomatic here, but I'd lean towards typing it `map[string]interface{}`. If we decide we want a concrete type in the future, it shouldn't be too difficult to change everywhere except in `Client4`.

**Comment by hmhealey** [comment]
Echoing my comment from https://github.com/mattermost/mattermost-redux/pull/1348 here for visibility:

We discussed this a bit more internally, and there's some bigger issues we discovered since changing the type in the redux state will break all plugins relying on the config only containing strings. We still need to figure out how what parts of mattermost-redux we want accessible by plugins and how we can do that to best ensure backwards compatibility, so we're going to close these tickets for the time being while we sort that all out. We don't have a specific timeline for that yet, but that will likely involve breaking changes, so it would be a good time to break this for as well.
